libraries/ms-drivers/test/test_ads1259_adc.c

```diff
@@ -0,0 +1,147 @@
+    if (cmd == 0x22) rx_data[0] = registers.CONFIG_2;
+    return STATUS_CODE_OK;
+  }
+  switch (cmd) {
+    // Commands used in ads1259_init()
+    case ADS1259_STOP_READ_DATA_CONTINUOUS:
+      LOG_DEBUG("STOP DATA CMD SENT\n");
+      return STATUS_CODE_OK;
+    case ADS1259_RESET:
+      LOG_DEBUG("RESET CMD SENT\n");
+      return STATUS_CODE_OK;
+    case (ADS1259_WRITE_REGISTER | ADS1259_ADDRESS_CONFIG0):
+      LOG_DEBUG("WRITE_REG CMD SENT\n");
+      TEST_ASSERT_TRUE(tx_len == 5 && rx_len == 0);
+      TEST_ASSERT_TRUE(tx_data[1] == 3);
+      registers.CONFIG_0 = tx_data[2];
+      registers.CONFIG_1 = tx_data[3];
+      registers.CONFIG_2 = tx_data[4];
+      break;
+    case ADS1259_OFFSET_CALIBRATION:
+      LOG_DEBUG("OFFSET CALIBRATION SENT\n");
+      registers.OFC_0 = 0;
+      registers.OFC_1 = 0;
+      registers.OFC_2 = 0;
+      break;
+    case ADS1259_GAIN_CALIBRATION:
+      LOG_DEBUG("GAIN CALIBRATION SENT\n");
+      registers.FSC_0 = 0;
+      registers.FSC_1 = 0;
+      registers.FSC_2 = 0;
+      break;
+    // Start of Ads1259_get_conversion_data()
+    case ADS1259_START_CONV:
+      LOG_DEBUG("CONVERSIONS STARTED \n");
+      break;
+    case ADS1259_READ_DATA_BY_OPCODE:
+      LOG_DEBUG("SETTING CONVERSION DATA\n");
+      rx_data[0] = 0xFF;
+      rx_data[1] = 0xFF;
+      rx_data[2] = 0xFF;
+      rx_data[3] = (uint8_t)((0x2FD + ADS1259_CHECKSUM_OFFSET) & 0x7F);
+      break;
+  }
+  return STATUS_CODE_OK;
+}
+
+static void prv_error_handler(Ads1259StatusCode code, void *context) {
+  if (code == ADS1259_STATUS_CODE_OUT_OF_RANGE) LOG_DEBUG("ADC READING IS OUT OF RANGE\n");
+  if (code == ADS1259_STATUS_CODE_CHECKSUM_FAULT) LOG_DEBUG("CHECKSUM INVALID\n");
+}
+
+void setup_test() {
+  Ads1259Settings settings = {
+    .spi_port = TEST_AD1259_SPI_PORT,
+    .spi_baudrate = TEST_BAUDRATE,
+    .mosi = TEST_MOSI_PIN,
+    .miso = TEST_MISO_PIN,
+    .sclk = TEST_SCLK_PIN,
+    .cs = TEST_CS_PIN,
+    .handler = prv_error_handler,
+  };
+  registers.CONFIG_0 = 0;
+  registers.CONFIG_1 = 0;
+  registers.CONFIG_2 = 0;
+  registers.OFC_0 = 1;
+  registers.OFC_1 = 1;
+  registers.OFC_2 = 1;
+  registers.FSC_0 = 1;
+  registers.FSC_1 = 1;
+  registers.FSC_2 = 1;
+  interrupt_init();
+  gpio_init();
+  soft_timer_init();
+  TEST_ASSERT_OK(ads1259_init(&settings, &storage));
+  TEST_ASSERT_EQUAL((registers.FSC_0 | registers.FSC_1 | registers.FSC_2), 0);
+  TEST_ASSERT_EQUAL((registers.OFC_0 | registers.OFC_1 | registers.OFC_2), 0);
+  TEST_ASSERT_EQUAL(registers.CONFIG_0, register_lookup[0]);
+  TEST_ASSERT_EQUAL(registers.CONFIG_1, register_lookup[1]);
+  TEST_ASSERT_EQUAL(registers.CONFIG_2, register_lookup[2]);
+}
+
+void teardown_test(void) {}
+
+void test_ads1259_init() {
+  uint32_t test_raw = 0xFFFFFF;
+  ads1259_get_conversion_data(&storage);
+  delay_ms(TEST_DATA_SETTLING_TIME_MS);
+  TEST_ASSERT_EQUAL(storage.conv_data.MSB | storage.conv_data.MID | storage.conv_data.LSB, 0xFF);
```

Comment:
our assertions are the other way around: the expected comes first, then the actual. Check the function parameters